### PR TITLE
Simple parse fix

### LINKED ISSUE #353: Smaller pow10
Using Peter's code we can use up less space in the implementation of pow10

### LINKED ISSUE #484: Parsing issue
Hello,

I'm trying to port my application ([Yadoms](https://github.com/Yadoms/yadoms)) which use rapidjson, to your new boost::json

I find a parsing issue when writing my unit tests.

I try to parse a decimal value (18.4 from my input string) and then retreive value as double.

My sample code

```c++
      const std::string data("{"
         "\"DecimalParameter\": 18.4"
         "}");

      auto& jv = boost::json::parse(data);
      boost::json::value *decimal = jv.get_object().if_contains("DecimalParameter");
      if (decimal)
      {
         BOOST_CHECK_EQUAL(decimal->as_double(), 18.4);
         BOOST_CHECK_EQUAL(decimal->get_double(), 18.4);
         BOOST_CHECK_EQUAL(decimal->to_number<double>(), 18.4);
         BOOST_CHECK_EQUAL(decimal->to_number<float>(), 18.4f);
      }
```

The results:

```
  Message: 
    check decimal->as_double() == 18.4 has failed [18.400000000000002 != 18.399999999999999]
    check decimal->get_double() == 18.4 has failed [18.400000000000002 != 18.399999999999999]
    check decimal->to_number<double>() == 18.4 has failed [18.400000000000002 != 18.399999999999999
```

The parsing is not exact, but works with floats!

Environnement:
 * boost : 1.75.0
 * VisualStudio 2019 16.8.3
 * build configuration x86

I mention also @sgallou (other main developer of Yadoms)

### PR DESCRIPTION
**Pull request by grisumbras**
Fixes #484. ~Also maybe fixes # 353.~

### CODE REVIEW

File: include/boost/json/basic_parser_impl.hpp [hunk 1]
```diff
@@ -144,14 +98,23 @@ dec_to_float(
...
-        return x * 1e-305 *
-            pow10(e+305);
-    return x * pow10(e);
+    {
+        e += 305;
+        x *= 1e-305;
+    }
+
+    if(e < -308)
+        return x * 0.0;
```
Comment (grisumbras) [inline include/boost/json/basic_parser_impl.hpp:112]: I don't know if division by infinity can trap on some platforms/configurations. If it doesn't, this check can be removed.

Comment (vinniefalco) [reply include/boost/json/basic_parser_impl.hpp:112]: I don't understand this line, why not just `return 0` ?

Comment (grisumbras) [reply include/boost/json/basic_parser_impl.hpp:112]: Are negative zeros important?

Comment (grisumbras) [reply include/boost/json/basic_parser_impl.hpp:112]: Actually, I think,  the whole block can be removed. This will result in `x / INFINITY`.

**Comment by grisumbras** [comment]
@vinniefalco this can be merged now.

**Comment by jano42** [comment]
@vinniefalco : thanks ;-)

**Pull request merged**